### ISSUE TITLE
What's the target of cache symlink supposed to be???

### ISSUE THREAD

**Issue by iionly** [Elgg/Elgg#10617]
I've tried setting up the cache symlink now on a 2.2 test installation and today also on a master test installation. Automatic symlink creation always fails... whatever...

Now a manual creation: on the 2.2 installation I EXPECTED (no info in docs and no help text available) that the target of the cache symlink is supposed to be the simple_cache folder. DID NOT WORK! Resulted in missing CSS. It seemed to work if the target of the cache link is the data directory itself. In this case the corresponding option on the advanced settings page is greyed out and the pages look right.

Now on master it says: "Correctly configured symbolic link must link /elgg/install/directory/cache/ to /path/to/datadir/views_simplecache/"

But it I do this I have again the same issue with CSS not getting loaded (not found) and most likely no other stuff cached either. If I set the symlink to point to the data directory, the corresponding option is not greyed out (so not active?).

So, it the cache symlink really working as it is supposed to work or it there some major issue somewhere? I have to say that I would have some major concerns about security on a production site if linking to the data directory and not the simplecache subfolder would really be the way it is supposed to work.

**Comment by hypeJunction** [comment]
Does your server config follow symlinks?

**Comment by iionly** [comment]
Symlinks support enabled (via Elgg's .htaccess).
"AllowOverride All" set for Elgg install directory in Apache config.

Nevertheless it does NOT work if link target is /path/to/datadir/views_simplecache/. I have to correct myself in one detail: it's not working on Elgg 2.2.3 (for whatever reason I thought I installed master but installed 2.2.3...).

What I have found out:
* it works (on Elgg 2.2.2) when also adding  -SymLinksIfOwnerMatch in .htaccess. Problem in my case (and likely in many other cases, too) is that not all directories within the path to the views_simplecache folder are owned by the user Apache runs as. But this seems to be required by Apache for the symlink to work correctly. I don't know if it's recommended to add -SymLinksIfOwnerMatch though.

* On 2.2.3 it does not work due to the change in _elgg_is_cache_symlinked() made in https://github.com/Elgg/Elgg/commit/b0958483211ab3a0a7e2678a43ffb860b3cebe2a. Here readlink() gives me an "Invalid argument" error. But maybe this can be fixed by backporting https://github.com/Elgg/Elgg/commit/4ca717d96bbb6782445ae326f6924dfcb0d4d550 to the 2.2 branch.

**Comment by iionly** [comment]
I figured out how to get it to work without using  -SymLinksIfOwnerMatch. It's only necessary that target file/folder and source of the symlink belong to the same owner. By using the "-h" option of chown I was able to change the owner of the cache link to the Apache user (in my case: chown -h wwwrun:www cache).

So, only the backport of https://github.com/Elgg/Elgg/commit/4ca717d96bbb6782445ae326f6924dfcb0d4d550 might be necessary - if there's a 2.2.4 release planned before releasing 2.3.0.
